storage/src/main/java/tech/pegasys/teku/storage/client/CombinedChainDataClient.java

```diff
@@ -333,10 +334,22 @@ public boolean isChainDataFullyAvailable() {
     return result;
   }
 
-  public UInt64 getBestSlot() {
+  public UInt64 getHeadSlot() {
     return this.recentChainData.getBestSlot();
```

Comment:
Should we apply this rename to `recentChainData.getBestSlot()` too? I must admit I'm getting confused about best vs head vs current slot and what they all mean.  It may be worth adding some JavaDoc to define them.

Reply:
Yeah, I've thought about doing this several times and just avoided it because of the noise, but worth doing here since its kind of topical.